program-runtime/src/invoke_context.rs

```diff
@@ -345,6 +345,7 @@ impl<'a> InvokeContext<'a> {
             .invoke_stack
             .last()
             .and_then(|frame| frame.program_id())
+            .cloned()
```

Comment:
In both `verify()` and `verify_for_update()` I've cloned the program_id pubkey, as to solve its ref-to-self issue. I am assuming this clone is non-consequential, from a performance view. If that's not the case though, let me know.